src/main/java/com/unityTest/courseManagement/entity/Comment.java

```diff
@@ -0,0 +1,60 @@
+package com.unityTest.courseManagement.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.hibernate.annotations.GenericGenerator;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Models a comment on a source item
+ */
+@Data
+@RequiredArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "COMMENT")
+public class Comment {
+
+	@Id
+	@Column(name = "ID")
+	@GeneratedValue(generator = "sequence-generator")
+	@GenericGenerator(
+		name = "sequence-generator",
+		strategy = "org.hibernate.id.enhanced.SequenceStyleGenerator",
+		parameters = {
+			@org.hibernate.annotations.Parameter(name = "sequence_name", value = "COMMENT_SEQUENCE"),
+			@org.hibernate.annotations.Parameter(name = "initial_value", value = "1000"),
+			@org.hibernate.annotations.Parameter(name = "increment_size", value = "1")})
+	private int id;
+
+	// Type of source being commented on
+	@Enumerated(EnumType.STRING)
+	@NotNull
+	@Column(name = "SOURCE_TYPE")
+	private SourceType sourceType;
+
+	// Id of source item being commented on
+	@NotNull
+	@Column(name = "ITEM_ID")
+	private int itemId;
```

Comment:
`itemId` is a bit ambiguous, why not:
```suggestion
	@NotNull
	@Column(name = "SOURCE_ID")
	private int sourceId;
```

or `parentId`. But I feel that `itemId` can be ambiguous if you are not familiar with the ER.

Reply:
Agreed. `itemId` is a bit ambigous, but I'd like to keep the idea that it's the item's id not the source's. How about `sourceItemId`?